smdebug_rulesconfig/actions/actions.py

```diff
@@ -0,0 +1,177 @@
+from smdebug_rulesconfig.actions.utils import validate_string
+
+
+class Action(object):
+    def __init__(self, **action_parameters):
+        """
+        Base class for action, which is to be invoked when a rule fires.. Offers `serialize` function to convert action
+        parameters to a string dictionary. This class is not meant to be initialized directly. Accepts dictionary of
+        action parameters and drops keys whose values are `None`.
+
+        :param action_parameters: Dictionary of action parameters.
+        """
+        action_parameters["name"] = self.__class__.__name__.lower()
+        self.action_parameters = {
+            key: value for key, value in action_parameters.items() if value is not None
+        }
+
+    def serialize(self):
+        """
+        Serialize the action parameters as a string dictionary.
+
+        :return: Action parameters serialized as a string dictionary.
+        """
+        return (
+            "{"
+            + ", ".join(
+                [f'\\"{key}\\": \\"{value}\\"' for key, value in self.action_parameters.items()]
+            )
+            + "}"
+        )
+
+
+class ActionList(object):
+    def __init__(self, *actions):
+        """
+        Higher level object to maintain a list of actions to be invoked when a rule is fired. Offers higher level
+        `serialize` function to handle serialization of actions as a string list of dictionaries.
+
+        :param actions: List of actions.
+        """
+        assert all(
```

Comment:
Good point, updated all assertions to to raise actual exceptions with the error message.

Again need to keep track of this since we don't have the same behavior across repos.